content/blog/2024-01-11-xilem-backend-roadmap.md

```diff
@@ -0,0 +1,157 @@
+
+This is good, but the "learning from the past" part takes a lot of time, and in the meantime, I'd argue that we've neglected end users a bit. We've abandonned Druid before Xilem was ready as an alternative, and then we worked on foundational and experimental projects that would make Xilem stronger and sounder in the long term, but left the project in an awkward state in the meantime (except for the web backend, which made a lot of progress).
+
+In the [ECS-backed GUI framework](https://www.leafwing-studios.com/blog/ecs-gui-framework/) article I linked, Bevy maintainers have this to say:
+
+> Not to be too harsh, but a lot of the existing Rust GUI solutions... just aren't very good. There's a lot of passable options, but they all have non-trivial drawbacks. No one has really risen to the top as a clear winner. [...] Deep down, we all know that we can do better, and we should.
+
+My ambition is to makes this paragraph obsolete before the end of the year. I want Xilem to get back to being suggested to newbies in the same breath as Iced and SlintUI. In the next few years, I want to entire ecosystem to get to a point where people talk about Rust GUI like they talk about ripgrep or rustls.
+
+To give one specific example: my personal goal is to implement seamless hot reloading for Xilem before the end of 2024. I believe it's feasible, and I believe doing it will radically change the way people think about Rust GUI, along with [other improvements I hope to unlock over time](https://poignardazur.github.io/2023/02/02/masonry-01-and-my-vision-for-rust-ui/).
+
+In the shorter term, my plan is to focus on basic features, widgets and styling options so people can experiment with Xilem, building on Raph's work on Vello over the past year. [John Skottis's Xilem UX Notes](https://gist.github.com/giannissc/172c6c591997ee14f6120134a7990697) give a good picture of the kinds of things I'll be working on.
+
+In the medium term, I also plan to work on the performance aspect of Xilem. People from Google Fonts have expressed interest in seeing how much performance we can get from Rust UI. While my mission isn't *limited* to performance work, it will be an ongoing focus, especially since it's an area where the Rust language brings unique opportunities. I'll go into more details on Rust performance in a later article.
+
+
+## Major changes I want to implement soon
+
+These are the big changes I want to work on soon. Ideally, some items of this list should be completed by the end of the month (but, you know, time estimates are hard).
+
+### Switching to Masonry as a backend
```

Comment:
This section is ok but may cause pushback. If I were writing it, I'd focus on the "merge" concept and say that the fastest way to get to the end goal is to merge things into the Masonry codebase. You can also crow about the advantages: the fact that the code base is in better shape, the value in having a separate crate less tied to the Xilem reactive layer, etc.

Keep in mind there is little disagreement on the goal within linebender, but the goal also deserves to be clearly communicated, especially as this reaches a broader audience. A lot of what we have in this section is process. You've got quite a bit of leeway there, as you'll be doing the work.